arcade/camera/camera_2d.py

```diff
@@ -0,0 +1,713 @@
+        self._projection.bottom = _bottom * self._data.zoom
+
+    @property
+    def projection_top(self) -> float:
+        """
+        The top edge of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS NOT scaled by zoom.
+        If this isn't what you want,
+        use projection_top_scaled instead.
+        """
+        return self._projection.top
+
+    @projection_top.setter
+    def projection_top(self, _top: float) -> None:
+        self._projection.top = _top
+
+    @property
+    def projection_top_scaled(self) -> float:
+        """
+        The top edge of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS scaled by zoom.
+        If this isn't what you want,
+        use projection_top instead.
+        """
+        return self._projection.top / self._data.zoom
+
+    @projection_top_scaled.setter
+    def projection_top_scaled(self, _top: float) -> None:
+        self._projection.top = _top * self._data.zoom
+
+    @property
+    def projection_near(self) -> float:
+        """
+        The near plane of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS NOT scaled by zoom.
+        """
+        return self._projection.near
+
+    @projection_near.setter
+    def projection_near(self, _near: float) -> None:
+        self._projection.near = _near
+
+    @property
+    def projection_far(self) -> float:
+        """
+        The far plane of the projection in world space.
+        This is not adjusted with the camera position.
+
+        NOTE this IS NOT scaled by zoom.
+        """
+        return self._projection.far
+
+    @projection_far.setter
+    def projection_far(self, _far: float) -> None:
+        self._projection.far = _far
+
+    @property
+    def viewport(self) -> Tuple[int, int, int, int]:
+        """
+        The pixel area that will be drawn to while the camera is active.
+        (left, right, bottom, top)
+        """
+        return self._projection.viewport
+
+    @viewport.setter
+    def viewport(self, _viewport: Tuple[int, int, int, int]) -> None:
+        self._projection.viewport = _viewport
+
+    @property
+    def viewport_width(self) -> int:
+        """
+        The width of the viewport.
+        Defines the number of pixels drawn too horizontally.
+        """
+        return self._projection.viewport[2]
+
+    @viewport_width.setter
+    def viewport_width(self, _width: int) -> None:
+        self._projection.viewport = (self._projection.viewport[0], self._projection.viewport[1],
+                                     _width, self._projection.viewport[3])
+
+    @property
+    def viewport_height(self) -> int:
+        """
+        The height of the viewport.
+        Defines the number of pixels drawn too vertically.
+        """
+        return self._projection.viewport[3]
```

Comment:
Same thing as above with the viewport height. This will fail if the viewport starts at anything other than the origin. I think it should be:

`return self._projection.viewport[3] - self._projection.viewport[1]`

Reply:
The viewport is stored as left, bottom, width, and height. So this is not necessary.